src/main/java/space/obminyashka/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -48,16 +49,20 @@ public class AdvertisementServiceImpl implements AdvertisementService {
     private String dateFormat;
 
     @Override
-    public List<AdvertisementTitleDto> findRandomNThumbnails(int amount, UUID excludeAdvertisementId, Long subcategoryId) {
-        final var totalRecordsSize = advertisementRepository.countByIdNotAndSubcategoryId(excludeAdvertisementId, subcategoryId);
-        final var bound = (int) (totalRecordsSize / amount);
-        final var randomPage = bound > 0 ? random.nextInt(bound) : 0;
-        return findAllThumbnails(excludeAdvertisementId, subcategoryId, PageRequest.of(randomPage, amount)).getContent();
+    public List<AdvertisementTitleDto> findRandomNThumbnails(AdvertisementFindThumbnailsRequest advertisementFindThumbnailsRequest) {
+        final var totalRecordsSize = advertisementRepository
+                .countByIdNotAndSubcategoryId(advertisementFindThumbnailsRequest.getExcludeAdvertisementId(), advertisementFindThumbnailsRequest.getSubcategoryId());
```

Comment:
Just make it shorted **advertisementFindThumbnailsRequest** variable name and if you need to split the code do it after the first bracket